simulator/mips/mips_instr.h

```diff
@@ -161,6 +161,7 @@ class BaseMIPSInstr
         static const MapType isaMapRI;
         static const MapType isaMapIJ;
         static const MapType isaMapMIPS32;
+        typename MapType::iterator find_entry( MapType isaMap, std::string_view name);
```

Comment:
While calling the function, you copy the `std::map` completely. Consider passing the constant reference instead.

Reply:
The function should return `MapType::const_iterator` as we do not intend to change the maps, and the function should have `const` qualifier as we do not change the object itself. So the signature would be:
```c++
typename MapType::const_iterator find_entry( MapType isaMap, std::srting_view name) const;
```

Also, a typedef would be useful here.